test/unit/argument_parser/format_parse_validators_test.cpp

```diff
@@ -691,16 +692,33 @@ TEST(validator_test, arithmetic_range_validator_error)
 
 TEST(validator_test, value_list_validator_success)
 {
+    // type deduction
+    // --------------
+    // all arithmetic types are deduced to double in order to easily allow chaining of arithmetic validators
```

Comment:
This explanation could also go into the class docs. I was actually wondering why it's `double` for all arithmetic types...